cmd/ecr-credential-provider/main.go

```diff
@@ -59,40 +71,108 @@ func defaultECRProvider(region string, registryID string) (*ecr.ECR, error) {
 	return ecr.New(sess), nil
 }
 
-func (e *ecrPlugin) GetCredentials(ctx context.Context, image string, args []string) (*v1.CredentialProviderResponse, error) {
-	registryID, region, registry, err := parseRepoURL(image)
+func publicECRProvider() (*ecrpublic.ECRPublic, error) {
+	// ECR public registries are only in one region and only accessible from regions
+	// in the "aws" partition.
+	sess, err := session.NewSessionWithOptions(session.Options{
+		Config:            aws.Config{Region: aws.String(ecrPublicRegion)},
+		SharedConfigState: session.SharedConfigEnable,
+	})
 	if err != nil {
 		return nil, err
 	}
 
+	return ecrpublic.New(sess), nil
+}
+
+type credData struct {
+	registry  string
+	authToken *string
+	expiresAt *time.Time
+}
+
+func (e *ecrPlugin) getPublicCredsData() (credData, error) {
+	klog.Infof("Getting creds for public registry")
+	var err error
+
+	if e.ecrPublic == nil {
+		e.ecrPublic, err = publicECRProvider()
+	}
+	if err != nil {
+		return credData{}, err
+	}
+
+	output, err := e.ecrPublic.GetAuthorizationToken(&ecrpublic.GetAuthorizationTokenInput{})
+	if err != nil {
+		return credData{}, err
+	}
+
+	if output == nil {
+		return credData{}, errors.New("response output from ECR was nil")
+	}
+
+	if output.AuthorizationData == nil {
+		return credData{}, errors.New("authorization data was empty")
+	}
+
```

Comment:
Let's add a check for `AuthorizationToken` is present as well?

(like the one we have in https://github.com/kubernetes/cloud-provider-aws/blob/52888eac04c58cda120c50f8225eb991395d9683/cmd/ecr-credential-provider/main.go#L91-L93)